server/base/src/main/java/org/apache/accumulo/server/conf/codec/GzipPropEncoding.java

```diff
@@ -0,0 +1,237 @@
+   *
+   * @param vProps
+   *          the versioned properties.
+   * @return a byte array with the serialized properties.
+   */
+  @Override
+  public byte[] toBytes(final VersionedProperties vProps) {
+
+    try (ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        DataOutputStream dos = new DataOutputStream(bos)) {
+
+      // write header - version id, isCompressed
+      encodingOpts.encode(dos);
+
+      // write updated property versioning info (data version, time stamp)
+      dos.writeInt(vProps.getNextVersion());
+      dos.writeUTF(vProps.getTimestampISO());
+
+      // write the property map keys, values.
+      if (encodingOpts.isCompressed()) {
+        writeMapCompressed(bos, vProps.getAllProperties());
+      } else {
+        writeMap(dos, vProps.getAllProperties());
+      }
+
+      dos.flush();
+
+      return bos.toByteArray();
+
+    } catch (IOException ex) {
+      throw new IllegalStateException("Encountered error serializing properties", ex);
+    }
+  }
+
+  @Override
+  public VersionedProperties fromBytes(final byte[] bytes) {
+
+    try (ByteArrayInputStream bis = new ByteArrayInputStream(bytes);
+        DataInputStream dis = new DataInputStream(bis)) {
+
+      // read header - version and isCompressed.
+      EncodingOptions opts = new EncodingOptions(dis);
+
+      // read versioning information
+      int dataVersion = dis.readInt();
+      Instant timestamp = tsFormatter.parse(dis.readUTF(), Instant::from);
+
+      // read the property map keys, values
+      Map<String,String> aMap;
+      if (opts.isCompressed()) {
+        aMap = readCompressedMap(bis);
+      } else {
+        aMap = readMap(dis);
+      }
+
+      return new VersionedPropertiesImpl(dataVersion, timestamp, aMap);
+
+    } catch (IOException ex) {
+      throw new UncheckedIOException("Encountered error deserializing properties", ex);
+    }
+  }
+
+  /**
+   * Read and uncompress an input stream compressed with GZip. The input stream is not closed by
+   * this method.
+   *
+   * @param is
+   *          an input stream
+   * @return a map with the property k.v pairs
+   * @throws IOException
+   *           if an exception occurs reading from the stream
+   */
+  private Map<String,String> readCompressedMap(final InputStream is) throws IOException {
+
+    try (GZIPInputStream gzipIn = new GZIPInputStream(is);
+        DataInputStream dis = new DataInputStream(gzipIn)) {
+      return readMap(dis);
+    }
+  }
+
+  /**
+   * Read the property map from a data input stream
+   *
+   * @param dis
+   *          a data input stream
+   * @return the property map
+   * @throws IOException
+   *           if an exception occurs reading from the stream.
+   */
+  private Map<String,String> readMap(DataInputStream dis) throws IOException {
+
+    Map<String,String> aMap = new HashMap<>();
+    int items = dis.readInt();
+
+    for (int i = 0; i < items; i++) {
+      Map.Entry<String,String> e = readKV(dis);
```

Comment:
`readKV` is only called from this location. You could just put the method body here instead of making a new method. Same comment applies to `writeKV`.